include/libtorrent/aux_/vector.hpp

```diff
@@ -52,14 +52,14 @@ namespace libtorrent { namespace aux {
 		{
 			TORRENT_ASSERT(idx >= IndexType(0));
 			TORRENT_ASSERT(idx < end_index());
-			return this->base::operator[](static_cast<underlying_index>(idx));
+			return this->base::operator[](static_cast<std::size_t>(idx));
```

Comment:
yeah, if ``deprecated-functions=on`` (which it is by default) the conversion operator is implicit, to be backwards compatible with using ``int`` everywhere